app/harvesters/hathi_harvester.rb

```diff
@@ -0,0 +1,94 @@
+require 'net/http'
+require 'rubygems/package'
+require 'zlib'
+
+
+##
+# A harvester for Hathi Trust MARC XML dump files.
+#
+# Supports either harvesting a previously downloaded dump file, or downloading
+# the latest available dump file and harvesting that.
+class HathiHarvester < Krikri::Harvesters::MarcXMLHarvester
+
+  ##
+  # @see Krikri::Harvesters::MarcXMLHarvester#each_collection
+  def each_collection
+    tarball_source do |tar_file|
+      each_tar_entry(tar_file) do |entry|
+        yield entry
+      end
+    end
+  end
+
+  private
+
+  ##
+  # @yield [IO] Give an IO stream containing an archive of MARC XML collection
+  #   files in TAR format.
+  def tarball_source(&block)
+    if File.exists?(uri)
+      File.open(uri, &block)
+    else
+      download_url(get_latest_tarball_url, &block)
+    end
+  end
+
+  ##
+  # @return [URI] Determine the URL of the most recent Hathi data dump.
+  def get_latest_tarball_url
+    html = Net::HTTP.get(URI(uri))
+    parsed_html = Nokogiri::HTML(html)
+
+    file_names = parsed_html.xpath('//a[starts-with(@href, "dpla")]').map { |a| a.get_attribute("href") }
```

Comment:
This is a great idea to pull the tarfile from the index page.

They're using Apache to serve the index with `mod_autoindex`.  We rely here on hyperlinks being constructed with the name of the file, without a path, like `href="dpla_full_20151001.tar.gz"`.  We could have to revise this some day if they change to another webserver, or to another means of generating the index page.  How about if we have an XPath condition using `contains()` that looks for "dpla_full" to avoid issues down the line if some new method starts adding a path to the file?